src/main/java/io/quarkus/gizmo2/impl/BlockCreatorImpl.java

```diff
@@ -902,6 +902,18 @@ private void doIf(final Expr cond, final Consumer<BlockCreator> whenTrue, final
         }
         doIfInsn(CD_void, cond, wt, wf);
     }
+    
+    private void doIfRef(final Expr ref, final Kind kind, final Consumer<BlockCreator> whenTrue, final Consumer<BlockCreator> whenFalse) {
```

Comment:
None of this should be necessary, the existing `if` code should be able to detect and optimize this case.

Reply:
Removed.